lib/api/multiObjectDelete.js

```diff
@@ -0,0 +1,307 @@
+        '',
+        errorXML,
+        '</DeleteResult>',
+    ];
+    if (quietSetting) {
+        // just return errors
+        return xml.join('');
+    }
+    const deletedXML = [];
+    deleted.forEach(objKey => {
+        deletedXML.push(
+            '<Deleted>',
+            '<Key>', escapeForXML(objKey), '</Key>',
+            '</Deleted>'
+        );
+    });
+    xml[2] = deletedXML.join('');
+    return xml.join('');
+}
+
+function _parseXml(xmlToParse, next) {
+    return parseString(xmlToParse, (err, result) => {
+        if (err || !result || !result.Delete) {
+            return next(errors.MalformedXML);
+        }
+        const json = result.Delete;
+        // not quiet is the default if nothing specified
+        const quietSetting = json.Quiet && json.Quiet[0] === 'true';
+        // format of json is {"Object":[{"Key":["test1"]},{"Key":["test2"]}]}
+        const objects = json.Object.map(item => item.Key[0]);
+        return next(null, quietSetting, objects);
+    });
+}
+
+/**
+ * multiObjectDelete - Delete multiple objects
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {object} request - http request object
+ * @param {object} log - Werelogs logger
+ * @param {function} callback - callback to server
+ * @return {undefined}
+ */
+export default
+function multiObjectDelete(authInfo, request, log, callback) {
+    log.debug('processing request', { method: 'multiObjectDelete' });
+    if (!request.post) {
+        return callback(errors.MissingRequestBodyError);
+    }
+    const md5 = crypto.createHash('md5')
+        .update(request.post, 'utf8').digest('base64');
+    if (md5 !== request.headers['content-md5']) {
+        return callback(errors.BadDigest);
+    }
+
+    const bucketName = request.bucketName;
+    const canonicalID = authInfo.getCanonicalID();
+    // track the error results for any keys with an error response
+    const errorResults = [];
+    // track keys that have been successfully deleted
+    const deleted = [];
+    // track keys that are still on track to be deleted
+    let inPlay = [];
+    let totalContentLengthDeleted = 0;
+
+    return async.waterfall([
+        function parseXML(next) {
+            return _parseXml(request.post, (err, quietSetting, objects) => {
+                if (err) {
+                    return next(err);
+                }
+                if (objects.length < 1 || objects.length > 1000) {
+                    return next(errors.MalformedXML);
+                }
+                return next(null, quietSetting, objects);
+            });
+        },
+        function checkPolicies(quietSetting, objects, next) {
+            // if request from account, no need to check policies
+            if (!authInfo.isRequesterAnIAMUser()) {
+                inPlay = objects.slice();
+                return next(null, quietSetting);
+            }
+
+            // TODO: once arsenal's extractParams is separated from doAuth
```

Comment:
I trust that an issue will be opened once this is merged so that we can track it afterwards and not forget about it.